tests/test_oic_provider_logout.py

```diff
@@ -0,0 +1,977 @@
+from urllib.parse import urlparse
+
+import pytest
+import responses
+
+from oic import rndstr
+from oic.exception import ParameterError
+from oic.oauth2.message import ErrorResponse
+from oic.oic import DEF_SIGN_ALG
+from oic.oic.consumer import Consumer
+from oic.oic.message import AuthorizationResponse
+from oic.oic.message import EndSessionRequest
+from oic.oic.provider import Provider
+from oic.utils.authn.authn_context import AuthnBroker
+from oic.utils.authn.client import verify_client
+from oic.utils.authn.user import UserAuthnMethod
+from oic.utils.authz import AuthzHandling
+from oic.utils.http_util import CookieDealer
+from oic.utils.http_util import Response
+from oic.utils.http_util import SeeOther
+from oic.utils.keyio import KeyBundle
+from oic.utils.keyio import KeyJar
+from oic.utils.keyio import keybundle_from_local_file
+from oic.utils.sdb import DictSessionBackend
+from oic.utils.sdb import session_get
+from oic.utils.userinfo import UserInfo
+
+__author__ = "roland hedberg"
+
+CONSUMER_CONFIG = {
+    "authz_page": "/authz",
+    "scope": ["openid"],
+    "response_type": ["code"],
+    "user_info": {"name": None, "email": None, "nickname": None},
+    "request_method": "param",
+}
+
+SERVER_INFO = {
+    "version": "3.0",
+    "issuer": "https://connect-op.heroku.com",
+    "authorization_endpoint": "http://localhost:8088/authorization",
+    "token_endpoint": "http://localhost:8088/token",
+    "flows_supported": ["code", "token", "code token"],
+}
+
+CLIENT_CONFIG = {"client_id": "number5", "config": {"issuer": SERVER_INFO["issuer"]}}
+
+CLIENT_CONFIG_2 = {"client_id": "client0", "config": {"issuer": SERVER_INFO["issuer"]}}
+
+CLIENT_SECRET = "abcdefghijklmnop"
+CLIENT_ID = "client_1"
+
+KC_SYM = KeyBundle(
+    [
+        {"kty": "oct", "key": CLIENT_SECRET, "use": "ver"},
+        {"kty": "oct", "key": CLIENT_SECRET, "use": "sig"},
+    ]
+)
+KC_SYM2 = KeyBundle(
+    [
+        {"kty": "oct", "key": "drickyoughurt", "use": "sig"},
+        {"kty": "oct", "key": "drickyoughurt", "use": "ver"},
+    ]
+)
+
+BASE_PATH = os.path.abspath(os.path.join(os.path.dirname(__file__), "data/keys"))
+KC_RSA = keybundle_from_local_file(
+    os.path.join(BASE_PATH, "rsa.key"), "RSA", ["ver", "sig"]
+)
+
+KEYJAR = KeyJar()
+KEYJAR[CLIENT_ID] = [KC_SYM, KC_RSA]
+KEYJAR["number5"] = [KC_SYM2, KC_RSA]
+KEYJAR[""] = KC_RSA
+KEYJAR["https://connect-op.heroku.com"] = KC_RSA
+
+
+USERDB = {
+    "user": {
+        "name": "Hans Granberg",
+        "nickname": "Hasse",
+        "email": "hans@example.org",
+        "verified": False,
+        "sub": "user",
+    },
+    "username": {
+        "name": "Linda Lindgren",
+        "nickname": "Linda",
+        "email": "linda@example.com",
+        "verified": True,
+        "sub": "username",
+        "extra_claim": "extra_claim_value",
+    },
+}
+
+URLMAP = {CLIENT_ID: ["https://example.com/authz"]}
+
+
+def dummy_template_renderer(template_name: str, context: Dict[str, str]) -> str:
```

Comment:
This should probably not be here. The default template rendered can be used. And if any modifications for proper logout functionality are needed, they should be implemented (and documented) there.

Reply:
OK